query_server/query/src/metadata.rs

```diff
@@ -166,6 +176,183 @@ impl MetaData for LocalCatalogMeta {
+            database_name = self.database_name.as_str()
+        }
+
+        match self.catalog.schema(database_name) {
+            None => Err(MetadataError::DatabaseNotExists {
+                database_name: database_name.to_string(),
+            }),
+            Some(_db_cfgs) => {
+                let schema = Arc::new(Schema::new(vec![Field::new(
+                    "Table",
+                    DataType::Utf8,
+                    false,
+                )]));
+
+                let tables = self.engine.list_tables(database_name).unwrap();
+                let batch = RecordBatch::try_new(schema, vec![Arc::new(StringArray::from(tables))])
+                    .unwrap();
+
+                let batches = vec![Arc::new(batch)];
+
+                Ok(Output::StreamData(stream_from_batches(batches)))
+            }
+        }
+    }
+
+    fn describe_database(&self, name: &str) -> Result<Output> {
+        match self.engine.get_db_schema(name) {
+            None => Err(MetadataError::DatabaseNotExists {
+                database_name: name.to_string(),
+            }),
+            Some(db_cfg) => {
+                let schema = Arc::new(Schema::new(vec![
+                    Field::new("TTL", DataType::Utf8, false),
+                    Field::new("SHARD", DataType::Utf8, false),
+                    Field::new("VNODE_DURATION", DataType::Utf8, false),
+                    Field::new("REPLICA", DataType::Utf8, false),
+                    Field::new("PRECISION", DataType::Utf8, false),
+                ]));
+
+                let ttl = db_cfg.config.ttl.to_string();
+                let shard = db_cfg.config.shard_num.to_string();
+                let vnode_duration = db_cfg.config.vnode_duration.to_string();
+                let replica = db_cfg.config.replica.to_string();
+                let precision = db_cfg.config.precision.to_string();
+
+                let batch = RecordBatch::try_new(
+                    schema,
+                    vec![
+                        Arc::new(StringArray::from(vec![ttl.as_str()])),
+                        Arc::new(StringArray::from(vec![shard.as_str()])),
+                        Arc::new(StringArray::from(vec![vnode_duration.as_str()])),
+                        Arc::new(StringArray::from(vec![replica.as_str()])),
+                        Arc::new(StringArray::from(vec![precision.as_str()])),
+                    ],
+                )
+                .unwrap();
+
+                let batches = vec![Arc::new(batch)];
+
+                Ok(Output::StreamData(stream_from_batches(batches)))
+            }
+        }
+    }
+
+    fn describe_table(&self, table: &str) -> Result<Output> {
+        let table_name;
+        let database_name;
+
+        if table.contains('.') {
+            (database_name, table_name) = table.split_once('.').unwrap();
+        } else {
+            table_name = table;
+            database_name = self.database_name.as_str();
+        }
```

Comment:
Parse the database name and table name from the string `xx.xx` or `xx`, could reference `TableReference`.

```Rust
    let table: TableReference = name.as_str().into();
    let catalog_name = catalog.catalog_name();
    let schema_name = catalog.schema_name();
    let table_ref = table.resolve(&catalog.catalog_name(), &catalog.schema_name());
```

Reply:
nice hint